### ISSUE TITLE
Cannot query S3 paths containing whitespace

### ISSUE THREAD

**Issue by andygrove** [apache/arrow-rs#2799]
**Describe the bug**

I cannot register a path containing a space.

```
create external table test2 stored as parquet location 's3://andygrove-benchmark-data/trip data/yellow_tripdata_2022-06.parquet';
ObjectStore(NotFound { path: "trip%20data/yellow_tripdata_2022-06.parquet"
```

It works if I change the path to not have spaces:

```
create external table test stored as parquet location 's3://andygrove-benchmark-data/trip_data/yellow_tripdata_2022-06.parquet';
0 rows in set. Query took 0.429 seconds.
```

**To Reproduce**

**Expected behavior**

**Additional context**

**Comment by andygrove** [comment]
@tustvold Here is another potential object store bug that I found

**Comment by tustvold** [comment]
This is sadly expected, https://docs.rs/object_store/latest/object_store/path/struct.Path.html#path-safety, I'm not really sure what can be done about this...

**Comment by andygrove** [comment]
We should add least add a check and throw an error? The S3 error response is somewhat obscure.

It will be a shame if DataFusion and Ballista cannot support querying certain public data sets like nyc-tlc because they have spaces in paths. Other query engines support it. One example: https://coiled.io/blog/nyc-taxi-parquet-csv-index-error/

I will plan on digging into this more soon to understand the issue better.

**Comment by tustvold** [comment]
> We should add least add a check and throw an error?

Agreed, I would have expected DataFusion to refuse that query as the parquet location is not a valid URL. Possibly ObjectStoreUrl is escaping rather than parsing.

> It will be a shame if DataFusion and Ballista cannot support querying certain public data sets

I welcome alternative suggestions for how to handle path escaping

**Comment by andygrove** [comment]
Perhaps some kind of user-provided config to override the default behavior? I will take a look this weekend.

**Comment by tustvold** [comment]
There is a somewhat related issue I ran into when playing with this, but you most definitely can create, read, etc... objects with paths containing spaces - https://github.com/apache/arrow-rs/issues/2800

**Comment by tustvold** [comment]
I think this was fixed by #2801

**Comment by alamb** [comment]
Migrating from arrow-rs issue [#2799](https://github.com/apache/arrow-rs/issues/2799)
